src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceBatchSource.java

```diff
@@ -79,7 +81,20 @@ public SalesforceBatchSource(SalesforceSourceConfig config) {
   @Override
   public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
     // validate when macros not yet substituted
-    config.validate(pipelineConfigurer.getStageConfigurer().getFailureCollector());
+    FailureCollector collector = pipelineConfigurer.getStageConfigurer().getFailureCollector();
+    if (!config.getConnection().canAttemptToEstablishConnection()) {
+      config.validateSOQLQueryParameters(collector);
```

Comment:
This looks like a new validation ?

Reply:
This is not a new validation. We need to perform this if connection is not there and still we have to check whether soql is correct and both of sObject and SOQL query are not null, one should have some value.